### DISCUSSION TITLE
How can I display the parameters of a failed DataAttribute test?

### DISCUSSION THREAD

**Discussion by MuddyWally** [xunit/xunit#2448] [Question] [answered]
Hello!
We use a very simple custom DataAttribute that simply gets one string parameter, the filename to read, parses the file and populates the `IEnumerable<object[]>`.
The attribure is marked with `AllowMultiple = true`, so we can have multiple files running with the same test.
But if there are for example 4 files and the test looks like this
```
        [FileData("TestData/Example1.par")]
        [FileData("TestData/Example2.par")]
        [FileData("TestData/Example3.par")]
        [FileData("TestData/ExampleEmpty.par")]
	public void SomeTest(List<int> items) {	…
```
it is not trivial to find out, due to which file the test failed, as the name of the file (i.e. the given parameter) is never mentioned in the error message.

Is there a way to show this information? Am I understanding something wrong?
How would I implement this with the current version?

Thanks in advance!

A Little Addendum:
The JSON example from [Andrew Lock](https://andrewlock.net/creating-a-custom-xunit-theory-test-dataattribute-to-load-data-from-json-files/) seems to suffer from the same problem. It's easy to use multiple JSON files on the same test, but if one fails, the search begins :-)

**Comment by bradwilson** [comment, marked as answer]
There's no direct way to do this, but you could update your [FileData] attribute to include the filename where the file came from, and update the test method's signature for that (like `public void SomeTest(string filename, List<int> items)`). Even though you would not use this filename parameter in your test, it _would_ show up in the display name of the test (like the list of ints does). It's a bit of a hack but should help you debug.

**Reply by MuddyWally** [reply]
Thanks!
That's a great idea and I can just pass the filename on, from the attribute function to the actual call.
